aspnetcore/performance/caching/middleware.md

```diff
@@ -1,48 +1,65 @@
+[!code-csharp[Main](middleware/samples/2.x/Program.cs?name=snippet1&highlight=4)]
 
-[!code-csharp[Main](middleware/sample/Startup.cs?name=snippet2)]
+# [ASP.NET Core 1.x](#tab/aspnetcore1x)
 
-The Response Caching Middleware only caches 200 (OK) server responses. Any other responses, including [error pages](xref:fundamentals/error-handling), will be ignored by the middleware.
+[!code-csharp[Main](middleware/samples/1.x/Startup.cs?name=snippet1&highlight=3)]
 
->[!WARNING]
+---
+
+Configure the app to use the middleware with the `UseResponseCaching` extension method, which adds the middleware to the request processing pipeline. The sample app adds a [`Cache-Control`](https://tools.ietf.org/html/rfc7234#section-5.2) header to the response that caches cacheable responses for up to 10 seconds. The sample sends a [`Vary`](https://tools.ietf.org/html/rfc7231#section-7.1.4) header to configure the middleware to serve a cached response only if the [`Accept-Encoding`](https://tools.ietf.org/html/rfc7231#section-5.3.4) header of subsequent requests matches that of the original request.
+
+# [ASP.NET Core 2.x](#tab/aspnetcore2x)
+
+[!code-csharp[Main](middleware/samples/2.x/Program.cs?name=snippet1&highlight=8)]
+
+# [ASP.NET Core 1.x](#tab/aspnetcore1x)
+
+[!code-csharp[Main](middleware/samples/1.x/Startup.cs?name=snippet2&highlight=3)]
+
+---
+
+The Response Caching Middleware only caches 200 (OK) server responses. Any other responses, including [error pages](xref:fundamentals/error-handling), are ignored by the middleware.
+
+> [!WARNING]
 > Responses containing content for authenticated clients must be marked as not cacheable to prevent the middleware from storing and serving those responses. See [Conditions for caching](#conditions-for-caching) for details on how the middleware determines if a response is cacheable.
 
 ## Options
-The middleware offers two options for controlling response caching.
+The middleware offers three options for controlling response caching.
 
-Option | Default Value
---- | ---
-UseCaseSensitivePaths | <p>Determines if responses will be cached on case-sensitive paths.</p><p>The default value is `false`.</p>
-MaximumBodySize | <p>The largest cacheable size for the response body in bytes.</p>The default value is `64 * 1024 * 1024` [64 MB (67,108,864 bytes)].</p>
+| Option                | Default Value |
+| --------------------- | ------------- |
+| UseCaseSensitivePaths | Determines if responses are cached on case-sensitive paths.</p><p>The default value is `false`. |
+| MaximumBodySize       | The largest cacheable size for the response body in bytes.</p>The default value is `64 * 1024 * 1024` [64 MiB]. |
```

Comment:
Yeah ... MiB *is* confusing ... correct ... but confusing. :smile:

Let's roll the 🎲 🎲 on just having the math there. I'll fix it now (and the \[]).